client/src/app/core/repositories/polls/poll-repository.service.ts

```diff
@@ -0,0 +1,383 @@
+        const result: (PollAction.AnalogOption & Identifiable)[] = [];
+        for (const option of pollOptions) {
+            if (!isUpdate) {
+                this.validateOption(option);
+            } else {
+                delete option.text;
+                delete option.content_object_id;
+            }
+            result.push({
+                id: option.id,
+                content_object_id: option.content_object_id,
+                text: option.text,
+                Y: toFixedPointNumber(option.Y),
+                A: toFixedPointNumber(option.A),
+                N: toFixedPointNumber(option.N)
+            });
+        }
+        return result;
+    }
+
+    private getNonAnalogOptions(pollOptions: any[]): PollAction.NonAnalogOption[] {
+        const result: PollAction.NonAnalogOption[] = [];
+        for (const option of pollOptions) {
+            this.validateOption(option);
+            result.push({
+                content_object_id: option.content_object_id,
+                text: option.text
+            });
+        }
+        return result;
+    }
+
+    private validateOption(option: PollAction.AnalogOption | PollAction.NonAnalogOption): void {
+        if ((option.text && option.content_object_id) || (!option.text && !option.content_object_id)) {
+            throw new Error('Exactly one of text or content_object_id has to be given!');
+        }
+    }
+
+    public async resetPoll(poll: Poll): Promise<void> {
+        const payload: PollAction.ResetPollPayload = { id: poll.id };
+        return this.sendActionToBackend(PollAction.RESET, payload);
+    }
+
+    public async anonymize(poll: Poll): Promise<void> {
+        const payload: PollAction.AnonymizePollPayload = { id: poll.id };
+        return this.sendActionToBackend(PollAction.ANONYMIZE, payload);
+    }
+
+    public async startPoll(poll: Poll): Promise<void> {
+        const payload: PollAction.StartPollPayload = { id: poll.id };
+        return this.sendActionToBackend(PollAction.START, payload);
+    }
+
+    public async stopPoll(poll: Poll): Promise<void> {
+        const payload: PollAction.StopPollPayload = { id: poll.id };
+        return this.sendActionToBackend(PollAction.STOP, payload);
+    }
+
+    public async publishPoll(poll: Poll): Promise<void> {
+        const payload: PollAction.PublishPollPayload = { id: poll.id };
+        return this.sendActionToBackend(PollAction.PUBLISH, payload);
+    }
+
+    public async updateOptionForPoll(poll: Poll, update: Partial<PollAction.OptionUpdatePayload>): Promise<void> {
+        if (poll.type === PollType.Analog) {
+            const payload: PollAction.OptionUpdatePayload = {
+                id: poll.id,
+                Y: update.Y,
+                N: update.N,
+                A: update.A
+            };
+            return this.sendActionToBackend(PollAction.UPDATE_OPTION, payload);
+        } else {
+            console.warn('Cannot update an option for a non-analog poll!');
```

Comment:
use early returns and an `throw new Error`. THis methos should not be callable by the any component for non-analog polls.